base/cvd/cuttlefish/host/commands/cvd/selector/instance_group_record.cpp

```diff
@@ -59,20 +59,54 @@ Result<LocalInstanceGroup> LocalInstanceGroup::Create(
 
   for (const auto& instance_proto : group_proto.instances()) {
     instances.emplace_back(group_proto, instance_proto);
-    ids.insert(instance_proto.id());
+    auto id = instance_proto.id();
+    if (id != 0) {
+      // Only non-zero ids are checked, zero means no id has been assigned yet.
+      CF_EXPECTF(ids.find(id) == ids.end(),
+                 "Instances must have unique ids, foud {} repeated", id);
+      ids.insert(id);
+    }
     names.insert(instance_proto.name());
   }
-  CF_EXPECT(ids.size() == (size_t)group_proto.instances_size(),
-            "Instances must have unique ids");
   CF_EXPECT(names.size() == (size_t)group_proto.instances_size(),
             "Instances must have unique names");
   return LocalInstanceGroup(group_proto, instances);
 }
 
+void LocalInstanceGroup::SetHomeDir(const std::string& home_dir) {
+  CHECK(group_proto_.home_directory().empty())
```

Comment:
Should these functions return a `Result`?

Reply:
A fail of this check represents a coding error. Normally I would try to make the compiler detect these, but in this case I can't, so a runtime abort is the next best thing.